test/unittest/xmlparser/XMLParserTests.cpp

```diff
@@ -622,15 +636,760 @@ TEST_F(XMLParserTests, DataBuffer)
 
 // INIT NACHO SECTION
 
-// FINISH NACHO SECTION	
+/*
+ * This test checks The return of the loadXMLProfiles method when a correct xml is parsed
+ */
+TEST_F(XMLParserTests, loadXMLProfiles)
+{
+
+    xmlparser::up_base_node_t root_node;
+    tinyxml2::XMLDocument xml_doc;
+    tinyxml2::XMLElement* titleElement;
+
+    const char * xml =
+    "\
+    <profiles>\
+        <publisher profile_name=\"test_publisher_profile\"\
+        is_default_profile=\"true\">\
+            <qos>\
+                <durability>\
+                    <kind>TRANSIENT_LOCAL</kind>\
+                </durability>\
+            </qos>\
+        </publisher>\
+        <subscriber profile_name=\"test_subscriber_profile\" is_default_profile=\"true\">\
+            <historyMemoryPolicy>PREALLOCATED_WITH_REALLOC</historyMemoryPolicy>\
+            <userDefinedID>13</userDefinedID>\
+            <entityID>31</entityID>\
+        </subscriber>\
+    </profiles>\
+    ";
```

Comment:
The following applies to all embedded xml code:
Insert two indentations of 4 spaces each to keep a common style